libp2p/Host.cpp

```diff
@@ -783,15 +807,21 @@ void Host::run(boost::system::error_code const& _ec)
     unsigned reqConn = 0;
     {
         RecursiveGuard l(x_sessions);
-        for (auto const& p : m_peers)
         {
-            bool haveSession = havePeerSession(p.second->id);
-            bool required = p.second->peerType == PeerType::Required;
-            if (haveSession && required)
-                reqConn++;
-            else if (!haveSession && p.second->shouldReconnect() &&
-                        (!m_netConfig.pin || required))
-                toConnect.push_back(p.second);
+            for (auto p = m_peers.cbegin(); p != m_peers.cend(); p++)
```

Comment:
You should do `++p` only unless you erased in this iteration. Now it'll skip the next one after erased one.

Reply:
Good catch